File: Semestre_2/Vision_Computadora/Entregas/Matlab/T2_1_Histograma_color_gris.m
close all;
clc;

[file1,path1] = uigetfile({'*.png';'*.jpg';'*.jpeg';'*.bmp';'*.gif'},'File Selector');

ruta1=strcat(path1,file1);

A=imread(ruta1);
A=im2double(A);

figure,
subplot(121);imshow(A); title('Imagen A');
subplot(122); imhist(A); title('Histograma de A');

% Histograma de imagen en gris
Agris = rgb2gray(A);
figure,
subplot(121);imshow(Agris); title('Imagen gris');
subplot(122); imhist(Agris); title('Histograma de A gris');

% Histograma por canal RGB
A_r=A(:,:,1);
A_g=A(:,:,2);
A_b=A(:,:,3);

figure;
subplot(232);imshow(A); title('Imagen A');
subplot(234);imhist(A_r); title('Histograma canal rojo');
subplot(235);imhist(A_g); title('Histograma canal verde');
subplot(236);imhist(A_b); title('Histograma canal azul');

File: Semestre_2/Vision_Computadora/Entregas/Python/T2_1_Histograma_color_gris.py
import matplotlib.pyplot as plt
from skimage.io import imread
from skimage.color import rgb2gray
from tkinter import Tk, filedialog

Tk().withdraw()
file_path = filedialog.askopenfilename(
    initialdir="C:\\Users\\death\\Documents\\Maestria\\MCIA\\Semestre_2\\Vision_Computadora\\img",
    filetypes=[("Images", "*.jpg;*.png;*.jpeg;*.bmp;*.gif")]
)

A = imread(file_path)

if A.shape[-1] == 4:
    A = A[..., :3]

A = A.astype(float) / 255.0

plt.figure()
plt.subplot(1,2,1); plt.imshow(A); plt.title('Imagen A'); plt.axis('off')
plt.subplot(1,2,2); plt.hist(A.ravel(), bins=256, color='gray'); plt.title('Histograma de A')

Agris = rgb2gray(A)

plt.figure()
plt.subplot(1,2,1); plt.imshow(Agris, cmap='gray'); plt.title('Imagen gris'); plt.axis('off')
plt.subplot(1,2,2); plt.hist(Agris.ravel(), bins=256, color='gray'); plt.title('Histograma de A gris')

A_r = A[..., 0]
A_g = A[..., 1]
A_b = A[..., 2]

plt.figure()
plt.subplot(2,3,2); plt.imshow(A); plt.title('Imagen A'); plt.axis('off')
plt.subplot(2,3,4); plt.hist(A_r.ravel(), bins=256, color='red'); plt.title('Histograma canal rojo')
plt.subplot(2,3,5); plt.hist(A_g.ravel(), bins=256, color='green'); plt.title('Histograma canal verde')
plt.subplot(2,3,6); plt.hist(A_b.ravel(), bins=256, color='blue'); plt.title('Histograma canal azul')
plt.show()
